pkg/skaffold/deploy/status_check.go

```diff
@@ -220,9 +225,34 @@ func newCounter(i int) *counter {
 	}
 }
 
-func (c *counter) markProcessed(err error) int {
+func (c *counter) markProcessed(err error) counter {
```

Comment:
what do you think about moving all these instance methods instead into functions, like the style of `append`? 
`func markProcessed(c *counter, err error) counter`? that will emphasize it even more that these are not state mutations.